feat(formula): align to template formula

noelmcloughlin opened the pull request:
<!--
Please fill in this PR template to make it easier to review and merge.

It has been designed so that a lot of it can be completed *after* submitting it,
e.g. filling in the checklists.

Notes:
1. Please keep the PR as small as is practicable; the larger a PR gets, the harder it becomes to review and the more time it requires to get it merged.
2. Similarly, please avoid PRs that cover more than one type; it should be a _bug fix_ *OR* a _new feature_ *OR* a _refactor_, etc.
3. Please direct questions to the [`#formulas` channel on Slack](https://saltstackcommunity.slack.com/messages/C7LG8SV54/), which is bridged to `#saltstack-formulas` on Freenode.
4. Feel free to suggest improvements to this template by reporting an issue or submitting a PR.  The source of this template is:
  - https://github.com/saltstack-formulas/.github/blob/master/.github/pull_request_template.md
-->

### PR progress checklist (to be filled in by reviewers)
<!-- Please leave this checklist for reviewers to tick as they work through the PR. -->

- [x] Changes to documentation are appropriate (or tick if not required)
- [x] Changes to tests are appropriate (or tick if not required)
- [ ] Reviews completed

---

### What type of PR is this?
<!-- Please tick each box that is relevant (after creating the PR). -->

#### Primary type
<!-- There really should be only *one* of these types ticked for each PR. -->

- [ ] `[build]`    Changes related to the build system
- [ ] `[chore]`    Changes to the build process or auxiliary tools and libraries such as documentation generation
- [ ] `[ci]`       Changes to the continuous integration configuration
- [x] `[feat]`     A new feature
- [ ] `[fix]`      A bug fix
- [ ] `[perf]`     A code change that improves performance
- [ ] `[refactor]` A code change that neither fixes a bug nor adds a feature
- [ ] `[revert]`   A change used to revert a previous commit
- [ ] `[style]`    Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc.)

#### Secondary type
<!-- Most PRs should include all of the following types as well. -->

- [ ] `[docs]`     Documentation changes
- [ ] `[test]`     Adding missing or correcting existing tests

### Does this PR introduce a `BREAKING CHANGE`?
<!-- If so, change the following to a `Yes` and explain what the breaking changes are. -->
<!-- If there are multiple breaking changes, list them all. -->

YES

### Related issues and/or pull requests
<!-- Please link any related issues/PRs here, especially any issues that are closed by this PR. -->



### Describe the changes you're proposing
<!-- A clear and concise description of what you have implemented. -->
<!-- Consider explaining each commit if they cover different aspects of the proposed changes. -->

There are about ten Jetbrains formulas in saltstack-formulas which need attention.  This formula is a template for all ten jetbrains formulas and has been migrated to align to template-formula. Travis CI tests are passing and this PR has been tested successfully on MacOS.

This PR will be followed by ten more PR so that all Jetbrains formulas are aligned and in CI/CD.


### Pillar / config required to test the proposed changes
<!-- Provide links to the SLS files and/or relevant configs (be sure to remove sensitive info). -->



### Debug log showing how the proposed changes work
<!-- Include a debug log showing how these changes work, e.g. using `salt-minion -l debug`. -->
<!-- Alternatively, linking to Kitchen debug logs is useful, e.g. via. Travis CI. -->
<!-- Most useful is providing a passing InSpec test, which can be used to verify any proposed changes. -->



### Documentation checklist
<!-- Please tick each box that is relevant (after creating the PR). -->

- [ ] Updated the `README` (e.g. `Available states`).
- [ ] Updated `pillar.example`.

### Testing checklist
<!-- Please tick each box that is relevant (after creating the PR). -->

- [ ] Included in Kitchen (i.e. under `state_top`).
- [ ] Covered by new/existing tests (e.g. InSpec, Serverspec, etc.).
- [ ] Updated the relevant test pillar.

### Additional context
<!-- Add any other context about the proposed changes here. -->

noelmcloughlin committed:
feat(formula): align to template formula

BREAKING CHANGE: Major refactor of formula to bring it in alignment with the
template-formula. As with all substantial changes, please ensure your
existing configurations work in the ways you expect from this formula.

noelmcloughlin committed:
feat(semantic-release): standardise for this formula

noelmcloughlin commented:
MacOS trace
```
local:
----------
          ID: intellij-macos-app-install-curl
    Function: file.directory
        Name: /tmp/epp/intellij
      Result: True
     Comment: Files cleaned from directory /tmp/epp/intellij
     Started: 00:14:55.436915
    Duration: 42.334 ms
     Changes:   
              ----------
              /tmp/epp/intellij/intellij-2020.1:
                  ----------
                  removed:
                      Removed due to clean
              removed:
                  - /tmp/epp/intellij/intellij-2020.1
----------
          ID: intellij-macos-app-install-curl
    Function: pkg.installed
        Name: curl
      Result: True
     Comment: All specified packages are already installed
     Started: 00:14:59.628188
    Duration: 10061.295 ms
     Changes:   
----------
          ID: intellij-macos-app-install-curl
    Function: cmd.run
        Name: curl -Lo /tmp/epp/intellij/intellij-2020.1 https://download.jetbrains.com/idea/ideaIC-2020.1.dmg
      Result: True
     Comment: Command "curl -Lo /tmp/epp/intellij/intellij-2020.1 https://download.jetbrains.com/idea/ideaIC-2020.1.dmg" run
     Started: 00:15:09.693581
    Duration: 1056668.818 ms
     Changes:   
              ----------
              pid:
                  42086
              retcode:
                  0
              stderr:
                    % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                                   Dload  Upload   Total   Spent    Left  Speed
                  
                    0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
                  100   138  100   138    0     0    801      0 --:--:-- --:--:-- --:--:--   802
                  
                    0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
                    0  600M    0  447k    0     0   340k      0  0:30:08  0:00:01  0:30:07  451k
                    0  600M    0  895k    0     0   380k      0  0:26:55  0:00:02  0:26:53  442k
                    0  600M    0 1471k    0     0   441k      0  0:23:13  0:00:03  0:23:10  489k
              
                    << CUT >>
              
                   99  600M   99  599M    0     0   582k      0  0:17:36  0:17:35  0:00:01  800k
                   99  600M   99  600M    0     0   582k      0  0:17:36  0:17:36 --:--:--  745k
                  100  600M  100  600M    0     0   582k      0  0:17:36  0:17:36 --:--:--  753k
              stdout:
----------
          ID: intellij-macos-app-install-checksum
    Function: module.run
        Name: file.check_hash
      Result: True
     Comment: onlyif condition is false
     Started: 00:32:46.364667
    Duration: 771.576 ms
     Changes:   
----------
          ID: intellij-macos-app-install-checksum
    Function: file.absent
        Name: /tmp/epp/intellij/intellij-2020.1
      Result: True
     Comment: State was not run because onfail req did not change
     Started: 00:32:47.136970
    Duration: 0.004 ms
     Changes:   
----------
          ID: intellij-macos-app-install-macpackage
    Function: macpackage.installed
        Name: /tmp/epp/intellij/intellij-2020.1
      Result: True
     Comment: IntelliJ IDEA CE.app installed
     Started: 00:32:47.137133
    Duration: 7183.909 ms
     Changes:   
              ----------
              installed:
                  - IntelliJ IDEA CE.app
----------
          ID: intellij-macos-app-install-macpackage
    Function: file.managed
        Name: /tmp/mac_shortcut.sh
      Result: True
     Comment: File /tmp/mac_shortcut.sh is in the correct state
     Started: 00:32:54.321359
    Duration: 37.875 ms
     Changes:   
----------
          ID: intellij-macos-app-install-macpackage
    Function: cmd.run
        Name: /tmp/mac_shortcut.sh
      Result: True
     Comment: Command "/tmp/mac_shortcut.sh" run
     Started: 00:32:54.359796
    Duration: 1193.777 ms
     Changes:   
              ----------
              pid:
                  44068
              retcode:
                  0
              stderr:
              stdout:
                  alias file IntelliJ IDEA CE alias of folder Desktop of folder nmcloughlin of folder Users of startup disk
----------
          ID: intellij-config-file-file-managed-environ_file
    Function: file.managed
        Name: /etc/defaults/intellij.sh
      Result: True
     Comment: File /etc/defaults/intellij.sh is in the correct state
     Started: 00:32:55.554266
    Duration: 1634.646 ms
     Changes:   

Summary for local
------------
Succeeded: 9 (changed=4)
Failed:    0
```

myii commented:
@noelmcloughlin I've noticed your `template-formula` conversion PRs and the new formulas that you've added -- I'll try to get around to reviewing/finalising them soon (i.e. GitHub settings).  Thanks for your patience.

myii commented on kitchen.yml:
```diff
@@ -0,0 +1,185 @@
+
+provisioner:
+  name: salt_solo
+  # log_level: debug
```
```suggestion
  log_level: debug
```

myii reviewed (commented):
Nice conversion, @noelmcloughlin.  Only got one line being highlighted, as shown inline.  Are there issues when debug logging is used?

* Checked using https://github.com/myii/ssf-formula/pull/181

myii commented:
Looking at https://github.com/saltstack-formulas/jetbrains-intellij-formula/pull/29#issuecomment-624856478, worth squashing some of the commits?

myii commented:
Actually, I only did the _automated_ tests so far.  I still need to look at the `README` and the `FORMULA` file.  Bit busy right now but I'll do this when I get a chance.

myii commented:
@noelmcloughlin That last commit message is getting caught by the linter.  I've checked the other two files and found that the `depth` is missing from the `README`:

```rst
.. contents:: **Table of Contents**
   :depth: 1
```

So to finish off, how about adding this last change and then squashing (some of) the commits to avoid the warnings in https://github.com/saltstack-formulas/jetbrains-intellij-formula/pull/29#issuecomment-624856478?

noelmcloughlin committed:
docs(readme): add depth one

noelmcloughlin commented:
Comments implemented
1. enable kitchen debug
2. depth=1
3. squashed commits/conflicts.

I updated all my PRs for 1&2 above - hopefully, they are good now too.  Thanks!!!

myii committed:
chore(kitchen): remove extra blank line

myii commented:
> Comments implemented
> 
> 1. enable kitchen debug
> 2. depth=1
> 3. squashed commits/conflicts.
> 
> 
> I updated all my PRs for 1&2 above - hopefully, they are good now too. Thanks!!!

@noelmcloughlin You're welcome, we're merged here.  There was an extra blank line that crept into `kitchen.yml` but I added a tiny commit to deal with that.

saltstack-formulas-travis commented:
:tada: This PR is included in version 1.0.0 :tada:

The release is available on [GitHub release](https://github.com/saltstack-formulas/jetbrains-intellij-formula/releases/tag/v1.0.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
